extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-workspace-service-impl.ts

```diff
@@ -71,8 +71,17 @@ export class K8sWorkspaceServiceImpl implements WorkspaceService {
   }
 
   public async getWorkspaceSettings(): Promise<WorkspaceSettings> {
-    console.log('workspaceService.getWorkspaceSettings() not supported');
-    return {};
+    const publicURI = this.env.getPluginRegistryURL();
+    const privateURI = this.env.getPluginRegistryInternalURL() || publicURI;
+
+    if (publicURI && privateURI) {
+      return {
+        CHE_PLUGIN_REGISTRY_URL: publicURI,
+        CHE_PLUGIN_REGISTRY_INTERNAL_URL: privateURI,
```

Comment:
If we call these keys without `CHE_` prefix, as `PLUGIN_REGISTRY_URL` and `PLUGIN_REGISTRY_INTERNAL_URL`, then we don't need any changes in `che-plugin-service.ts`. Isn't it?

Reply:
> PLUGIN_REGISTRY_URL in che-plugin-service.ts means a bit different

Actually, it means the same - URL of the plugin registry. It doesn't matter how this registry is implemented and where it's located, etc.

`workspaceService.getWorkspaceSettings()` allows to abstract from where it's stored: either Che-Server WS settings or DevWorkspace env. variables.

So, if we get rid of the `CHE_` prefixes, we can make it more transparent in `che-plugin-service.ts`.

Reply:
looks like the returned object has the same key but with different values ?

Reply:
> looks like the returned object has the same key but with different values ?

I would say different keys with the same meaning.